### PR TITLE
Add Push To Hub functionality for Lightweight GAN

### PR DESCRIPTION
**Pull request by nateraw**
This PR aims to add the ability to push lightweight GAN to the hub periodically every `save_every` steps. It will also include misc bugfixes as I run across them.

### CODE REVIEW

**Comment by nateraw** [comment]
So looks like its working now. What isn't working is continuing from one of these checkpoints...but you can _load_ the model directly into the `LightweightGAN` object from the hub or from a local save file that was saved with `save_pretrained`. 

A good tester command:
`accelerate launch cli.py --dataset_name huggan/pokemon --push_to_hub --name lightweight-gan-pokemon --save_every 10 --organization_name huggan`

**Comment by nateraw** [comment]
Here's a screenshot of the above command. You can see its pushing every `save_every` steps.

Screen Shot 2022-04-07 at 10 17 50 PM

File: huggan/pytorch/lightweight_gan/cli.py [hunk 1]
```diff
@@ -100,7 +101,16 @@ def train_from_folder(
...
     wandb = False,
+    push_to_hub = False,
+    organization_name = None,
 ):
+    if push_to_hub:
+        if name == 'default':
+            raise RuntimeError(
+                "You've chosen to push to hub, but have left the --name flag as 'default'."
+                " You should name your model something other than 'default'!"
+            )
```
Comment (NielsRogge) [inline huggan/pytorch/lightweight_gan/cli.py:112]: Should this also raise an error if no `organization_name` is specified?

In the other scripts, this defaults to "huggan"

File: huggan/utils/hub.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from huggingface_hub import HfFolder, whoami
+
+def get_full_repo_name(model_id: str, organization: Optional[str] = None, token: Optional[str] = None):
+    if token is None:
+        token = HfFolder.get_token()
+    if organization is None:
+        username = whoami(token)["name"]
+        return f"{username}/{model_id}"
+    else:
+        return f"{organization}/{model_id}"
```
Comment (NielsRogge) [inline huggan/utils/hub.py:12]: I see you're pushing by default to the username, fine for me. 

In the other scripts, I set `organization_name` to "huggan" by default.

**Review by NielsRogge — APPROVED** [review]
LGTM, thanks for improving!

**Pull request merged**